substrate/client/network/transactions_v2/transactions_v2_proposal.md

```diff
@@ -0,0 +1,201 @@
+
+The response must include transaction bodies in the same order they were requested. If a transaction body cannot be delivered, an indicator that the transaction was not available should be included in its place.
+
+If the requested transaction is not available in the local pool the requesting peer's reputation shall be decreased. However, if the peer requests the transaction whose identifier was previously gossiped to that peer and transaction is found to be unknown (e.g. due to finalization or being dropped) its reputation shall remain unaffected. This kind of race condition is possible and there is little that can be done to prevent it.
+
+Once transaction bytes are downloaded the transaction should be sent to pool for further validation and processing.
+
+If a remote node is unable to provide a transaction it previously announced through identifiers flooding, the requesting node shall decrease its reputation. As noted in the previous paragraph, such situations may occur but should not be common. Reputation of nodes that gossip identifiers without being able to provide the corresponding transaction bodies should be decreased.
+
+The request protocol supports batch acquisition of transactions by accepting a `Vec<TxIdentifier>`.
+
+
+_Notes_:
+- In theory, the transaction body could be fetched only if there is an available space in transaction pool. Transaction shall not be silently dropped. On the other hand the gossiped transaction identifier may correspond to transaction with higher priority and we should submit such transaction immediately as it may be evicting some other lower-priority transactions. To avoid downloading same transaction multiple times, the implementation shall keep track of downloaded transactions. A txpool API could be leveraged for this (see implemntation notes).
+- The node could retain the bodies of transactions that were broadcasted and then dropped from the pool. This would prevent its reputation from being decreased due to an inability to provide a transaction body when the pool is under high pressure and operating at its limits.
+
+#### [Optional] Transaction Data Low-Fanout (`TxLF`).
+The low-fanout strategy is an optional, easily achievable enhancement aimed at improving transaction propagation latency and network resilience.
+
+The transaction descriptors containing full transaction data are relayed to a small number of (randomly / based on reputation) selected peers. This approach is taken in [etheruem protocol](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#transaction-exchange). When a transaction is submitted via RPC to the local node, the full transaction data may be relayed to an increased number of peers, as no other nodes initially possess this data. This approach helps decrease propagation time by ensuring quicker dissemination.
```

Comment:
I don't get this. So the low fanout is described as relaying full txs to a small number of randomly/reputation based selected peers (similarly to ETH). But then you say that when submitted via RPC to the local node, the full tx may be relayed to an increased number of peers - how is this different from the low fanout? Is it different when the tx is submitted via RPC to the local node? Tbh I am not sure which is the code path for this - does it involve the network, or is it mostly happening on the same machine.

Reply:
Bad wording. Entire paragraph is referring to the same thing.

Basically RPC will low-fanout transaction body to some peers (increased ==  small number of (randomly / based on reputation) selected peers ).


RPC is mentioned because it is the point of entry for transaction. And no other node has seen given tx so far. `TxLF` may help with better dissemination.